### PR TITLE
Feature/implement oc tabbed

### PR DESCRIPTION
**Pull request by marcus-herrmann**
## Description

Implements new accessible tab component `oc-tabbed`for Files sidebar. Related PR in ODS: https://github.com/owncloud/owncloud-design-system/pull/539

## Related Issue

- Fixes https://github.com/owncloud/owncloud-design-system/issues/535

## Motivation and Context
See motivation of ODS ticket: Aiming to improve screen reader and keyboard accessibility of tabs in Files Sidebar

## Types of changes
Changed FileDetails.vue to match with new tab component DOM

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>PrivateLinks</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>RestrictSharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>



























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>















































</p></details>

**Comment by marcus-herrmann** [comment]
I assume the tests are failing because of the new DOM structure. Since I cannot get the tests to run, could anybody assist me in remedying?

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPermissionsUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>



















































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPermissionsGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>



















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>



































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Resharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingAutocompletion</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>























































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>iPhone</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>



































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>



































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6609/

<p>






























</p></details>

**Comment by marcus-herrmann** [comment]
@PVince81 PR is rebased now

**Comment by PVince81** [comment]
you should now be able to rebase and then update the design system package so that the logic can work

**Comment by marcus-herrmann** [comment]
@PVince81 Like I wrote, I just rebased this morning. To my grasp I did update ODS (see: owncloud/owncloud-design-system#535 but it needs review first), or do you mean something else regarding these topics, for example having phoenix's package.json point to the ODS commit needed for this to run without merging this very PR?

File: apps/files/src/components/FileDetails.vue [hunk 1]
```diff
@@ -49,8 +59,25 @@ export default {
     close () {
       this.$emit('reset')
     },
+    handleActiveTab (payload) {
+      this.activeTab = this.activeTabFromComponent = payload
+    },
     showSidebar (app) {
       this.activeTab = app
+<<<<<<< HEAD
+=======
```
Comment (PVince81) [inline apps/files/src/components/FileDetails.vue]: conflicts

**Comment by PVince81** [comment]
@marcus-herrmann if you run `yarn upgrade owncloud-design-system` it should grab the latest and yarn.lock will be modified. I don't see a modified yarn.lock in this PR, so CI will not see it either and tests will fail.

Also see conflicts that need resolution

File: apps/files/src/components/FileDetails.vue (hunk 1 above)
Comment (marcus-herrmann) [reply apps/files/src/components/FileDetails.vue]: Resolved

**Comment by marcus-herrmann** [comment]
> @marcus-herrmann if you run yarn upgrade owncloud-design-system it should grab the latest and yarn.lock will be modified. I don't see a modified yarn.lock in this PR, so CI will not see it either and tests will fail.

> Also see conflicts that need resolution

@PVince81  I see. I thought this would update ODS to the latest commit on its master, which would be of no use since the related ODS-PR is not merged yet, thus not on master. Anyway: new yarn.lock now part of this very PR

**Comment by PVince81** [comment]
> which would be of no use since the related ODS-PR is not merged yet

I merged it yesterday, that's why you can update now

**Comment by marcus-herrmann** [comment]
> I merged it yesterday, that's why you can update now

Ah, received too many GitHub mails to realize this. Sorry. Thanks! Done.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>PrivateLinks</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6876/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>PrivateLinks</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>RestrictSharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6876/

<p>



























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>RestrictSharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>



























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFilePermissionsGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>



































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6876/

<p>























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingAcceptShares</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>























































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>







































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingFolderPermissionsGroups</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>



















































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPermissionsUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>



















































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingAutocompletion</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>























































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Resharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>















































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6876/

<p>





























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>iPhone</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>









































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>









































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>































































































































































































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6877/

<p>





























</p></details>

**Comment by PVince81** [comment]
@individual-it can someone help to adjust the tests ?

**Comment by individual-it** [comment]
Assigned myself for fixing the tests

**Comment by individual-it** [comment]
- [ ] issue: "NEW" - menu does not close after selecting any of the options
menu-not-hidden

**Comment by individual-it** [comment]
- [ ] issue: browser hangs when opening a folder while the "NEW" menu is open and then accessing that menu again inside of the folder
hang

**Comment by kiranparajuli589** [comment]
- [ ] issue: - more results option in auto-completion doesn't work properly
Screenshot from 2019-12-12 15-34-39

**Comment by PVince81** [comment]
@kiranparajuli589 please rebase. There was a problem on master with share autocomplete "more results" which is solved now.

this PR is only about tab control so any other issue is likely from somewhere else

**Comment by pascalwengerter** [comment]
Closed since we've introduced a different approach to the files sidebar and are not using the `oc-tabbed` component in web right  now

**Pull request closed without merging**